bionic/core/flow_execution.py

```diff
@@ -56,7 +60,9 @@ def __init__(self, bootstrap, flow_instance_uuid, task_key_logger):
         self._entries_by_task_key = {}
         self._pending_entries = []
         self._in_progress_entries = {}
-        self._blockage_lists_by_blocking_tk = defaultdict(list)
+        # TODO This is a set but _in_progress_entries is a dict. Could they both be
+        # sets?
+        self._blocked_entries = set()
```

Comment:
This might be my gap in python knowledge, but doesn't all elements of a set need to be hashable? Would entries always be hashable, even when the state or the cached values aren't?

Reply:
By default, Python classes are hashable and the hash function is based on their `id`. (However, some classes, like mutable containers, explicitly disable their hash function.) For our purposes, I think the default behavior should be fine. I do prefer to have a clearer hash key, but it didn't seem worth the trouble of keying on `entry.state.task_keys[0]`. WDYT? Also, in the future, when every entry has exactly one task key, the balance might shift a little.

Reply:
TIL!

If it works, I think it's fine. I don't think we are ever going to disable hash function for entries.